### ISSUE TITLE
OpenBLAS 0.3.8 issue on AMD Threadripper cpu

### ISSUE THREAD

**Issue by shyrma** [deeplearning4j/deeplearning4j#8747]
OpenBLAS 0.3.8 crashes on AMD Threadripper 3970X cpu, with SIGILL error code.
We always got following crash while testing TFGraphTestAllHelper.log_determinant.rank3:

- A fatal error has been detected by the Java Runtime Environment:
-  SIGILL (0x4) at pc=0x00007f86f33181ff, pid=130947, tid=0x00007f8839009700
- C  [libopenblas_nolapack.so.0+0x10a61ff]  sgemm_kernel_direct+0x126f

This issue is a blocker for us, since we basically unable to use devbox for java tests

**Comment by saudet** [comment]
I'm sure OpenBLAS 0.3.7 does the same. Work around that with OPENBLAS_CORETYPE, the same as was done for issue #4287.

**Comment by raver119** [comment]
re: 0.3.7. I don't think so, we've been using this box for 3 months now, and got into this problem only after update to 0.3.8 i think. It's pretty new issue there.

re: OPENBLAS_CORETYPE - nope, doesn't help.

**Comment by raver119** [comment]
@saudet you're right, 0.3.7 reproduces the same issue. So probably this crash is caused by the graph being added to resources.

I'll close this issue, we'll sort it out in some other way i think now.

**Comment by saudet** [comment]
Setting the OPENBLAS_CORETYPE environment variable to "Athlon" or "Core2" doesn't help? That's strange. That should disable pretty much all optimizations. Anyway, one more thing to try is to use MKL instead. The presets for OpenBLAS can load it with the system properties given here:
https://github.com/bytedeco/javacpp-presets/tree/master/openblas#documentation

**Comment by raver119** [comment]
Athlon doesn't help. We'll try Core2 now

**Comment by shyrma** [comment]
I've just checked, same situation - no success unfortunately

**Comment by saudet** [comment]
This is probably the cause: https://github.com/xianyi/OpenBLAS/issues/2526

**Comment by raver119** [comment]
Who's building OpenBLAS for JavaCPP?

**Comment by saudet** [comment]
The changes from commit https://github.com/bytedeco/javacpp-presets/commit/cb0031563cb2e5d0dcf7d22791ba4fcec7ce827c have been deployed.
Please give it a try with 0.3.9-1.5.4-SNAPSHOT: http://bytedeco.org/builds/

**Comment by AlexDBlack** [comment]
ND4J tests (including those that were consistently crashing on this system) are confirmed passing on 3990x with Linux (ubuntu 20.04) with the following version modifications:
https://github.com/KonduitAI/deeplearning4j/compare/master...KonduitAI:ab_openblas_test?expand=1

**Comment by raver119** [comment]
Everything is fine on 3950 and 3970 as well 👍
